TourMate/TourMate/Frontend/User/Services/FirebaseUserService.swift

```diff
@@ -6,15 +6,15 @@
 //
 
 import Foundation
-import FirebaseAuth
 
 struct FirebaseUserService: UserService {
     private let userRepository = FirebaseRepository(collectionId: FirebaseConfig.userCollectionId)
+    @Injected(\.authenticationManager) var authenticationManager: AuthenticationManager
 
     private let userAdapter = UserAdapter()
 
     func addUser(_ user: User) async -> (Bool, String) {
-        guard let currentUser = Auth.auth().currentUser,
+        guard let currentUser = authenticationManager.getCurrentFirebaseUser(),
```

Comment:
I think its better if we access the firebaseuserservice instead to check whether the user is logged in, and let the auth service be the only one accessing the auth manager.